modules/apps/portal/portal-verify-test/src/testIntegration/java/com/liferay/portal/verify/test/VerifyUUIDTest.java

```diff
@@ -99,8 +98,7 @@ public void testVerifyModel() {
 				expectedLog = "Invalid column name 'Unknown'.",
 				expectedType = ExpectedType.PREFIX
 			)
-		},
-		level = "ERROR", loggerClass = ThrowableAwareRunnable.class
+		}
```

Comment:
Hi @dantewang @yuhai ,
As what I can see, LogAssertionTestRule is not working at all. 
You can just modify the expected log message in this test class, and the test still pass. Please take a look at this issue.

Thanks.
Tina.

Reply:
Hi @tinatian, in the case, it doesn't use [LogAssertionTestRule.endAssert()](https://github.com/dantewang/liferay-portal/blob/20ccf46ffa0883a7b46e772a88a7005ca35fb027/portal-test-integration/src/com/liferay/portal/test/rule/LogAssertionTestRule.java#L69-L85) to assert because it doesn't print error log in loggerClass (I check the progress and no classes print the error log), instead of throwing the exception and then [_verifyException()](https://github.com/dantewang/liferay-portal/blob/20ccf46ffa0883a7b46e772a88a7005ca35fb027/modules/apps/portal/portal-verify-test/src/testIntegration/java/com/liferay/portal/verify/test/VerifyUUIDTest.java#L263-L317) use  @ExpectedLogs annotation to assert. 

I do the test by adding print the error log in VerifyUUID class. I can see the same behavior (test failed) between before fix and after fix. I might not understand your idea. Could you please explain it again?

Thanks,
Hai

Reply:
@yuhai,
`it doesn't print error log in loggerClass` This is the problem, if you know there is no error log, why you add annotation to the test method to assert the logging?
Either there is something wrong with the test method (you should not add those annotations), or there is something wrong with the test rule (you should assert if you have annotation but you did not print anything, this is a kind of failure too).

Tina

Reply:
@tinatian
1. I believe the test rule is working as designed. For example a test case will be executed on all kinds of databases and it's expected that sometimes some error log may be printed on, let's say, Sybase and DB2. These error logs are random and of no harm. Then only patterns of these logs should be annotated, and the test should pass with no log printed or with matching log printed.

2. This specific test case is a special one. As Hai said, the test throws exception instead of printing log. The the code uses the message patterns in the annotations to compare with the exception it catches and decide whether to pass. The usage here is more of a trick to take the convenience of the annotations, not really intentional to use the test rule.